Add feature to exclude datatypes

jack-gitdev opened the pull request.

jack-gitdev committed:
Add feature to exclude datatypes

FineAndDandy commented on warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java:
```diff
@@ -746,7 +746,32 @@ protected void loadQueryParameters(ShardQueryConfiguration config, Query setting
+            }
+
+            if (!excludeSet.isEmpty()) {
+                if (typeFilter.isEmpty()) {
```
throw an error if your ignoring due to the typeFilter being non-empty

FineAndDandy reviewed (changes requested):
can you add a unit test that covers the new functionality?

ivakegg commented on warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java:
```diff
@@ -746,7 +746,32 @@ protected void loadQueryParameters(ShardQueryConfiguration config, Query setting
+                    MetadataHelper metadataHelper = prepareMetadataHelper(config.getClient(), this.getMetadataTableName(), config.getAuthorizations(), config.isRawTypes());
+
+                    try {
+                        for (Type<?> type : metadataHelper.getAllDatatypes()) {
```
These are the wrong data types.  I know this is highly confusing, but these data types are the types assigned to fields (e.g. LcNoDiacritics).  Unfortunately we chose the initial "datatype" designation unwisely.  So "datatype" really needs a qualifier of some sort.  Field datatype vs event or ingest datatype ?  IngestDataTypes is used for various method parameters in the MetadataHelper.   Anyway, the method metadataHelper.getDatatypes((Set<String>)null) will get the data types you are looking for.

jack-gitdev committed:
Add unit tests for exclude data types feature

FineAndDandy commented on warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java:
```diff
@@ -746,7 +746,30 @@ protected void loadQueryParameters(ShardQueryConfiguration config, Query setting
+                    }
+                }
+
+                typeFilter.removeAll(excludeSet);
```
consider adding a test that there is at least one dataType left?

jack-gitdev commented:
Need to rework unit tests and add data to testing environment. Current data set does not support testing of necessary edge cases.

jack-gitdev committed:
Added data to the test data set, adjusted ShardQueryLogicTest to support the expectation of no results, and adjusted unit tests affected by the addition of new data.

jack-gitdev committed:
Merge branch 'integration' into exclude-types-feature

jack-gitdev commented:
Added data to the test data set, adjusted ShardQueryLogicTest to support the expectation of no results, and adjusted unit tests affected by the addition of new data. Removed draft status.

jack-gitdev committed:
Merge branch 'integration' into exclude-types-feature

jack-gitdev committed:
Merge branch 'integration' into exclude-types-feature

jack-gitdev committed:
Merge branch 'integration' into exclude-types-feature

jack-gitdev committed:
Merge branch 'integration' into exclude-types-feature

jack-gitdev committed:
Merge branch 'integration' into exclude-types-feature

jack-gitdev committed:
Merge branch 'integration' into exclude-types-feature
